assistant/assistant.go

```diff
@@ -0,0 +1,105 @@
+package assistant
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf/config"
+	"golang.org/x/net/websocket"
+)
+
+// Assistant represents a client that will connect to the cloud server to allow
+// for plugin management remotely.
+type Assistant struct {
+	Config *config.Config // stores plugins and agent config
```

Comment:
Probably want to create a custom `AssistantConfig` with things only necessary for the Assistant
Maybe things such as the endpoint to hit, retryInterval (for retrying when it cannot connect to the cloud server), etc..

Reply:
I agree that an AssistantConfig struct should be created, but I think having the  Agent Config is good. I haven't taken a long look at it's full functionality, but I know that the Agent Config contains all the plugins at the least and we can add and take out plugins. While we can build that into the AssistantConfig as well, code is not DRY then because we'll just be copying the code from AgentConfig.